docs/proposals/kubernetes-mcpgroup-crd.md

```diff
@@ -0,0 +1,272 @@
+# MCPGroup CRD for Kubernetes Operator
+
+## Problem Statement
+
+The CLI supports runtime groups for organizing MCP servers, but this is missing in Kubernetes. The Virtual MCP Server feature (PR #2106) requires groups to discover and aggregate backend servers. Without groups, Kubernetes users cannot use the Virtual MCP or organize their servers logically.
```

Comment:
I'm not sure if we need `MCPGroup` for this? We can use label/annotations nicely for the MCPServers? This would allow being a part of multiple groups too without having to manage overhead of another CRD and controllers

Reply:
You're right. The main use-case is for meta-mcp and virtual mcp.